### PR TITLE
Add Gnielinski's correlation for the Nusselt Number

### PR DESCRIPTION
**Pull request by luizaldeia**
Refs #24340

### CODE REVIEW

File: modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/bib/thm.bib [hunk 1]
```diff
@@ -83,4 +83,14 @@ @techreport{wolf_mccarthy1960
 institution = {Rocketdyne},
 number = {RR-60-12},
 year = {1960}
-}
\ No newline at end of file
+}
+
+@book{todreas2021nuclear,
```
Comment (licharlot) [inline modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/bib/thm.bib:88]: I think you already added it in another PR.

File: modules/thermal_hydraulics/src/materials/ADWallHeatTransferCoefficientGnielinskiMaterial.C [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      "cp", FlowModelSinglePhase::SPECIFIC_HEAT_CONSTANT_PRESSURE, "Specific heat of the fluid");
+  params.addParam<MaterialPropertyName>(
+      "mu", FlowModelSinglePhase::DYNAMIC_VISCOSITY, "Dynamic viscosity of the fluid");
+  params.addParam<MaterialPropertyName>(
+      "k", FlowModelSinglePhase::THERMAL_CONDUCTIVITY, "Heat conductivity of the fluid");
+  params.addParam<MaterialPropertyName>(
+      "T", FlowModelSinglePhase::TEMPERATURE, "Fluid temperature");
+  params.addParam<MaterialPropertyName>("T_wall", FlowModel::TEMPERATURE_WALL, "Wall temperature");
+  params.addClassDescription(
+      "Computes wall heat transfer coefficient for water using the Gnielinski correlation");
```
Comment (licharlot) [inline modules/thermal_hydraulics/src/materials/ADWallHeatTransferCoefficientGnielinskiMaterial.C:39]: Not just for water

File: modules/thermal_hydraulics/src/materials/ADWallHeatTransferCoefficientGnielinskiMaterial.C [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+void
+ADWallHeatTransferCoefficientGnielinskiMaterial::computeQpProperties()
+{
+  ADReal Pr = THM::Prandtl(_cp[_qp], _mu[_qp], _k[_qp]);
+  ADReal Re = std::max(1.0, THM::Reynolds(1., _rho[_qp], _vel[_qp], _D_h[_qp], _mu[_qp]));
+
+  if (Re < 1000)
+  {
+    mooseError("Gnielinski's correlation will produce negative heat transfer coefficients if the "
+               "Reynolds number is below 1000, please verify your assumptions");
```
Comment (licharlot) [inline modules/thermal_hydraulics/src/materials/ADWallHeatTransferCoefficientGnielinskiMaterial.C:67]: Make this a warning. Sometimes, we start a simulation with low velocities to help with convergence, but may still want to use that correlation.

File: modules/thermal_hydraulics/src/materials/ADWallHeatTransferCoefficientGnielinskiMaterial.C [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  else if (Re < 2300 || Re > 5E+6 || Pr < 0.5 || Pr > 2000)
+  {
+    mooseDoOnce(mooseWarning(
+        "The Gnielinski correlation is valid when Pr is between 0.5 and 2000, and Re is "
+        "between 2300 and 5000000.  Be aware that using values out of this range may lead to "
+        "significant errors in your results!"));
+  }
+
+  ADReal f = std::pow(1.82 * std::log10(Re) - 1.64, -2.0);
+  ADReal Nu = ((f / 8.0) * (Re - 1000.0) * Pr) /
```
Comment (licharlot) [inline modules/thermal_hydraulics/src/materials/ADWallHeatTransferCoefficientGnielinskiMaterial.C:78]: use Max( 0,Re-1000) per comment above.

**Review by licharlot — CHANGES_REQUESTED** [review]

File: modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/bib/thm.bib (hunk 1 above)
Comment (luizaldeia) [reply modules/thermal_hydraulics/doc/content/modules/thermal_hydraulics/bib/thm.bib:88]: Yes, I added in the htc_core branch that got merged, but I think I wrote this file before the merging, so I added it to compile the documentation, I will remove it

File: modules/thermal_hydraulics/src/materials/ADWallHeatTransferCoefficientGnielinskiMaterial.C (hunk 2 above)
Comment (luizaldeia) [reply modules/thermal_hydraulics/src/materials/ADWallHeatTransferCoefficientGnielinskiMaterial.C:39]: done

File: modules/thermal_hydraulics/include/materials/ADWallHeatTransferCoefficientGnielinskiMaterial.h [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+
+/**
+ * Computes wall heat transfer coefficient for liquid sodium using Schad-modified correlation
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/include/materials/ADWallHeatTransferCoefficientGnielinskiMaterial.h:15]:
```suggestion
 * Computes wall heat transfer coefficient for gases and water using the Gnielinski correlation
```

**Review by joshuahansel — CHANGES_REQUESTED** [review]
Just the small change that I see. Site link isn't working, so I can't see the rendered doc, but text looks ok.

**Review by joshuahansel — APPROVED** [review]

**Review by licharlot — APPROVED** [review]

**Pull request merged**